### PR TITLE
Release notebook for Numba 0.52

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Might need to mess about with the container to get graphviz to work?

**Comment by stuartarchibald** [comment]
CC @luk-f-a @StanczakDominik was wondering if either of you would like to do an initial review? No obligation/pressure :) [Direct rendered file link](https://github.com/numba/numba-examples/blob/ecc767a9a2d1c4b8d352e2515dfe21a7d61c4a37/notebooks/Numba_052_release_demo.ipynb) Thanks!

File: notebooks/Numba_052_release_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+    "%timeit fastmath_jit_sum_list(nb_list) # \"fastmath\" JIT function on typed list"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "CFG inspection enhancements\n",
+    "=========================\n",
+    "The Numba dispatcher's [`inspect_cfg()` method](https://numba.readthedocs.io/en/stable/reference/jit-compilation.html#Dispatcher.inspect_cfg) has been enhanced with colorized output and support for Python code interleaving to provide a more visual way to debug/tune code. For a more advanced demonstration, this feature is used in [the notebook](https://mybinder.org/v2/gh/numba/numba-examples?filepath=notebooks%2FNumba_052_refpruner.ipynb) explaining the new reference count pruning pass. A quick demonstration of this feature:"
```
Comment (sklam) [inline notebooks/Numba_052_release_demo.ipynb:121]: The inspect_cfg doc needs to be updated https://github.com/numba/numba/blame/484925571c7caaadd2170434ea889afed788a6f3/docs/source/reference/jit-compilation.rst#L195

File: notebooks/Numba_052_release_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    "    except Exception: # inf and nan not accepted\n",
+    "        caught = True\n",
+    "\n",
+    "    # String literal dtypes\n",
+    "    ones, zeros, empty = (np.ones((5,), 'int8'), np.zeros((3,), 'complex128'),\n",
+    "                          np.empty((0,), 'float32'))\n",
+    "\n",
+    "    return farr, split, arr_split, arr_contains, caught, ones, zeros, empty\n",
+    "    \n",
+    "demo_numpy()"
```
Comment (sklam) [inline notebooks/Numba_052_release_demo.ipynb:196]: The output is a bit hard to read given there are so many. I'd prefer it goes like:

```python
print(f"farr: {farr}\nsplit: {split}...")
```

**Comment by luk-f-a** [comment]
a few observations:
- the links to other notebooks on binder return "bad request" (maybe a temporary problem because this is not merged?)
-  I cannot run the inspect_cfg example. After installing `graphviz` and `python-graphviz` is still says 
```
FileNotFoundError: [Errno 2] No such file or directory: 'dot'

During handling of the above exception, another exception occurred:
...


...
ExecutableNotFound: failed to execute ['dot', '-Tdot_json'], make sure the Graphviz executables are on your systems' PATH
```
even though `which dot` returns the right path.

- I agree with @sklam's coment about the output of demo_numpy.

**Comment by stuartarchibald** [comment]
Thanks for the review @luk-f-a 

> a few observations:
> 
>     * the links to other notebooks on binder return "bad request" (maybe a temporary problem because this is not merged?)

Should be temporary, once merged the pages will be live and links should work.

>     * I cannot run the inspect_cfg example. After installing `graphviz` and `python-graphviz` is still says
> 
> 
> ```
> FileNotFoundError: [Errno 2] No such file or directory: 'dot'
> 
> During handling of the above exception, another exception occurred:
> ...
> 
> 
> ...
> ExecutableNotFound: failed to execute ['dot', '-Tdot_json'], make sure the Graphviz executables are on your systems' PATH
> ```
> 
> even though `which dot` returns the right path.

hmmmm, will try and fix up the container so this works on binder. Not sure why `dot` isn't working, might be that `dot_json` is too new a feature? Which version of `dot` do you have?

>     * I agree with @sklam's coment about the output of demo_numpy.

will fix!

**Review by sklam — APPROVED** [review]
Looks good now.
Getting this merged will be easier to tell if the container will work.

**Comment by stuartarchibald** [comment]
Thanks for the reviews, merging now to see if we need to do anything to the containers to get the CFG stuff working.

**Comment by luk-f-a** [comment]
> hmmmm, will try and fix up the container so this works on binder. Not sure why `dot` isn't working, might be that `dot_json` is too new a feature? Which version of `dot` do you have?


`dot - graphviz version 2.42.3 (20191010.1750)`

**Pull request merged**